src/System/Hapistrano.hs

```diff
@@ -330,6 +331,13 @@ restartServerCommand = do
     Nothing -> return "No command given for restart action."
     Just cmd -> runCommand (host conf) cmd
 
+cleanBuildScript :: [String] -> [String]
+cleanBuildScript allScriptLines = filter (not . isCommentOrEmpty) allScriptLines
+  where
+    isCommentOrEmpty line = isEmpty line || isComment line
```

Comment:
Does this work if there is leading space on the function?

We may want to talk about what the code is doing here. There may be a better approach, perhaps using megaparsec's code to ignore comments, or by implementing this in a different way altogether.